[DOCS-12260] Cloudcraft Azure support

git-thuerk-done opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?

Breaks out the prerequisites section into 2 tabs:
- AWS (no new content)
- Azure (all new content)
- Updates screenshots and videos throughout to reflect new Azure tab
- Adds new copy to the Group by and group by tags section to reflect new functionality for labels and service tags

### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it receives the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [ ] Ready for merge

**For Datadog employees**:

Your branch name MUST follow the `<name>/<description>` convention and include the forward slash (`/`). Without this format, your pull request will not pass CI, the GitLab pipeline will not run, and you won't get a branch preview. Getting a branch preview makes it easier for us to check any issues with your PR, such as broken links.

If your branch doesn't follow this format, rename it or create a new branch and PR.

[6/5/2025] Merge queue has been disabled on the documentation repo. If you have write access to the repo, the PR has been reviewed by a Documentation team member, and all of the required checks have passed, you can use the **Squash and Merge** button to merge the PR. If you don't have write access, or you need help, reach out in the #documentation channel in Slack.

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->

jamesponddotco commented:
From the Cloudcraft team, this looks good. I'd just update this part:

```markdown
These diagrams visually represent AWS cloud resources, allowing you to explore and analyze your environments.
```

I'd update it to include Azure instead of mentioning AWS only.

git-thuerk-done commented:
> ```
> AWS
> ```

Thanks @jamesponddotco good catch! Just merged that update.

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -42,37 +45,73 @@ Cloudcraft's core functionality is its ability to generate detailed architecture
+You can group resources by AWS and Azure tags, such as app, service, team, or cost center, to organize your view by team or workload. When grouping by tags, color-coded labels are displayed on each group. When grouping by the `service` tag, a raised block is displayed to visually indicate the service grouping.
 
-**Note**: Grouping by tags is supported for AWS tags only. Tags from the Datadog Agent (for example ,`env`, or `team` tags from local configuration) are not supported.
+**Note**: Grouping by tags is supported for AWS and Azure tags only. Tags from the Datadog Agent (for example ,`env`, or `team` tags from local configuration) are not supported. 
```
```suggestion
**Note**: Grouping by tags is supported for AWS and Azure tags only. Tags from the Datadog Agent (for example, locally configured `env` or `team` tags) are not supported. 
```

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -89,28 +128,25 @@ To apply a saved view to your diagram:
-Click the **\+Filter** menu to quickly filter your resources by commonly used tags such as service, team, region, and more. Additionally, click the **More Filters** option to filter by AWS tags, custom tags, and terraform tags. The filter option reloads the diagram to display only the infrastructure that matches the filter criteria.
-
-{{< img src="datadog_cloudcraft/cloudcraft_filter.png" alt="Filter feature in Cloudcraft" >}}
+Click the **\+Filter** menu to quickly filter your resources by commonly used tags such as service, team, region, and more. Additionally, click the **More Filters** option to filter by AWS and Azure tags, custom tags, and terraform tags. The filter option reloads the diagram to display only the infrastructure that matches the filter criteria.
```
```suggestion
Click the **\+Filter** menu to quickly filter your resources by commonly used tags such as service, team, region, and more. Additionally, click the **More Filters** option to filter by AWS and Azure tags, custom tags, and Terraform tags. The filter option reloads the diagram to display only the infrastructure that matches the filter criteria.
```

buraizu commented on content/en/datadog_cloudcraft/overlays.md:
```diff
@@ -92,19 +92,19 @@ By default, the security overlay shows Critical, High, and Medium severity misco
 You can view sensitive data matches for your S3 buckets. Click a resource with matches to learn more about the bucket's sensitive data matches. Then, hover over a filename and click its **Inspect in AWS** button.
 
-{{< img src="datadog_cloudcraft/overlays/cloudcraft_security_overlay_sensitive_data.png" alt="Security overlay in Cloudcraft with the sensitive data filter applied, highlighting the collapsible legend in the bottom left of the screen, and the Inspect in AWS button on the side panel." style="width:100%;" >}}
+{{< img src="datadog_cloudcraft/overlays/cloudcraft_security_overlay_sensitive_data_2.png" alt="Security overlay in Cloudcraft with the sensitive data filter applied, highlighting the collapsible legend in the bottom left of the screen, and the Inspect in AWS button on the side panel." style="width:100%;" >}}
```
Update to match the UI

buraizu commented on content/en/datadog_cloudcraft/overlays.md:
```diff
@@ -92,19 +92,19 @@ By default, the security overlay shows Critical, High, and Medium severity misco
 You can click on any resource that has findings to open a side panel with more details, allowing deeper investigation without leaving the diagram. Click **Investigate** to get more context about the finding and learn how to remediate it.
 
-{{< img src="datadog_cloudcraft/overlays/cloudcraft_security_overlay_misconfigurations.png" alt="Security overlay in Cloudcraft with the misconfigurations filter applied, highlighting the collapsible legend in the bottom left of the screen, and the Investigate button on the side panel." style="width:100%;" >}}
+{{< img src="datadog_cloudcraft/overlays/cloudcraft_security_overlay_misconfigurations_2.png" alt="Security overlay in Cloudcraft with the misconfigurations filter applied." style="width:100%;" >}}
```
```suggestion
{{< img src="datadog_cloudcraft/overlays/cloudcraft_security_overlay_misconfigurations_2.png" alt="Security overlay in Cloudcraft with the misconfigurations filter applied in the collapsible legend in the bottom of the screen, and highlighting the Investigate button" style="width:100%;" >}}
```

buraizu commented on content/en/datadog_cloudcraft/overlays.md:
```diff
@@ -123,7 +123,7 @@ Clicking a resource opens a detailed side panel with:
 This enables faster, in-context cost optimization without switching views.
 
-{{< img src="datadog_cloudcraft/cloudcraft_ccm_overlay_3.png" alt="Cloud Cost Management(CCM) overlay in Cloudcraft, highlighting the collapsible savings legend at the bottom left hand side of the screen." style="width:100%;" >}}
+{{< img src="datadog_cloudcraft/overlays/cloudcraft_ccm_overlay_4.png" alt="Cloud Cost Management(CCM) overlay in Cloudcraft, highlighting the collapsible savings legend at the bottom left hand side of the screen." style="width:100%;" >}}
```
```suggestion
{{< img src="datadog_cloudcraft/overlays/cloudcraft_ccm_overlay_4.png" alt="Cloud Cost Management (CCM) overlay in Cloudcraft, highlighting the recommendations option in the collapsible legend at the bottom left hand side of the screen." style="width:100%;" >}}
```

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -42,37 +45,73 @@ Cloudcraft's core functionality is its ability to generate detailed architecture
+**Note**: Grouping by tags is supported for AWS and Azure tags only. Tags from the Datadog Agent (for example ,`env`, or `team` tags from local configuration) are not supported. 
 
-{{< img src="datadog_cloudcraft/group_by_tag.mp4" alt="Group by tag feature in Cloudcraft, grouping by Team and Cost Center" video=true >}}
+{{< img src="datadog_cloudcraft/cloudcraft_group_by_with_team_tags.png" alt="Group by tag feature in Cloudcraft, grouping by Team" >}}
```
```suggestion
{{< img src="datadog_cloudcraft/cloudcraft_group_by_with_team_tags.png" alt="Cloudcraft landing page with Group by highlighted, and grouping by Team" >}}
```

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -42,37 +45,73 @@ Cloudcraft's core functionality is its ability to generate detailed architecture
+Enable the **Show All Controls** toggle to display the available **Group By** options. You can remove specific groupings by unchecking options like VPC and Region. To view the current nesting structure and add the Network ACL (Network Access Control List) layer, click the **+ Tags** menu.
 
-{{< img src="datadog_cloudcraft/cloudcraft_group_by_with_ccm.png" alt="Group by feature in Cloudcraft, highlighting the Group By menu" >}}
+{{< img src="datadog_cloudcraft/cloudcraft_group_by_with_network_acl.png" alt="Group by feature in Cloudcraft, highlighting the Group By menu, and the teams tag" >}}
```
```suggestion
{{< img src="datadog_cloudcraft/cloudcraft_group_by_with_network_acl.png" alt="Cloudcraft landing page with Group by highlighted, and the teams tag" >}}
```

buraizu replied:
I'm not seeing the teams tag in this screenshot. Does the image need to be updated, or the alt text?

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -42,37 +45,73 @@ Cloudcraft's core functionality is its ability to generate detailed architecture
+{{< img src="datadog_cloudcraft/getting_started.png" alt="Getting started in Cloudcraft, displaying a list of resources for the selected account and region" style="width:100%;" >}}
 
-**Note**: The account name in the **Account** dropdown originates from your AWS account tags in the AWS integration tile.
+<div class="alert alert-tip">The account name in the <strong>Account</strong> dropdown comes from your AWS account tags in the AWS integration tile. For Azure, the <strong>Azure Subscription</strong> name comes from the subscription name in your Azure integration tile's list of managed subscriptions.
```
```suggestion
<div class="alert alert-tip">The account name in the <strong>Account</strong> dropdown comes from your AWS account tags in the AWS integration tile. For Azure, the <strong>Subscription</strong> name comes from the subscription name in your Azure integration tile's list of managed subscriptions.
```

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -19,19 +19,22 @@ Cloudcraft offers a powerful, live read-only visualization tool for cloud archit
 - Accelerate incident MTTR and proactive governance tasks by simplifying infrastructure navigation.
 
-{{< img src="datadog_cloudcraft/cloudcraft_with_ccm_2.mp4" alt="Cloudcraft in Datadog video" video=true >}}
+{{< img src="datadog_cloudcraft/cloudcraft_with_azure_tab_2.mp4" alt="Cloudcraft in Datadog video" video=true >}}
```
```suggestion
{{< img src="datadog_cloudcraft/cloudcraft_with_azure_tab_2.mp4" alt="Video showing a Cloudcraft diagram in the security overlay. A resource is selected, opening a side panel with its details. The Changes option is selected from the left hand side of the side panel. In the Security section of the side panel, several misconfigurations are listed. The Investigate button next to one of the misconfigurations is clicked, opening a new side panel with details of the misconfiguration and a Next Steps section with options for Triage, Remediation, and More Actions" video=true >}}
```

buraizu reviewed (changes requested):
Looks good overall, but left a few suggestions and posted some questions.

git-thuerk-done commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -42,37 +45,73 @@ Cloudcraft's core functionality is its ability to generate detailed architecture
+  1. Navigate to [**Integrations > Azure**][15].
+  2. Select the App Registration that contains your Azure subscription.
+  3. On the Resource Collection tab, enable the **Resource Collection** toggle.
+  4. Optionally, if the subscription hasn't been added to Datadog yet, select **+ Add New App Registration** to add the account to Datadog first.
```
hey that's a great callout, Step 4 should absolutely be before step 2 for users that don't have subscriptions added yet.

git-thuerk-done commented:
Thanks @buraizu ! I incorporated your feedback in these 2 commits:
[One](https://github.com/DataDog/documentation/pull/32424/commits/012dc359d057e54c9cd397df27b4fa4cee3da4e3)
[Two](https://github.com/DataDog/documentation/pull/32424/commits/440c0e3972629d2d01dbdd7cc06ebd012f5e6744)
For the 2nd commit, you'll notice I also moved up the [requirements for Azure to](https://github.com/DataDog/documentation/pull/32424/commits/440c0e3972629d2d01dbdd7cc06ebd012f5e6744#diff-ccb799d3720ce01267ab240d8023f83a7c652e42dc58c3aaf5deabc8ec0a628aR66) above the resource collection section

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -19,60 +19,102 @@ Cloudcraft offers a powerful, live read-only visualization tool for cloud archit
+  1. Navigate to [**Integrations > Azure**][15].
+  2. Add your Azure subscription by selecting **+ Add New App Registration** if not already added.
+  3. Optionally, select the App Registration containing your Azure subscription.
+  4. On the Resource Collection tab, enable the **Enable Resource Collection** toggle.
```
```suggestion
  4. On the Resource Collection tab, ensure the **Enable Resource Collection** toggle is enabled.
```

buraizu commented on content/en/datadog_cloudcraft/_index.md:
```diff
@@ -19,60 +19,102 @@ Cloudcraft offers a powerful, live read-only visualization tool for cloud archit
+- Enable [resource collection][14] for your Azure accounts:
+  1. Navigate to [**Integrations > Azure**][15].
+  2. Add your Azure subscription by selecting **+ Add New App Registration** if not already added.
+  3. Optionally, select the App Registration containing your Azure subscription.
```
Could you confirm if step 3 is optional? It wasn't listed as optional before, and it looks like an app registration is always "selected" in the integration tile (when the integration is configured):

<img width="630" height="477" alt="azure-int-tile" src="https://github.com/user-attachments/assets/30972a5a-f1c0-4987-aeb0-8e82d33b6252" />

git-thuerk-done replied:
Oh sorry I think this is confusing the way it's written - My understanding is there are two paths to Enabling Resource collection for Azure 
- One path is if you don't already have any subscriptions added to the integration, they need to click Add New App Registration
- The other path is if they already have subscriptions added to the integration, they would click the app registration they previously enabled then click the Resource collection tab. 

I can see how this is confusing! Let me rephrase this. Either way you are correct, it is indeed not an optional step :) Thanks for flagging this.

buraizu reviewed (commented):
Thanks for applying that feedback! Just reiterated a previous suggestion with some additional context, and I also posted another question about the updated Azure steps
